packages/ui/src/components/va-virtual-scroller/useVirtualScrollerSizes.ts

```diff
@@ -0,0 +1,78 @@
+import { PropType, ExtractPropTypes, ref, Ref, shallowRef, computed, watch, onMounted } from 'vue'
+
+import { useResizeObserver } from '../../composables'
+
+export const useVirtualScrollerSizesProps = {
+  horizontal: { type: Boolean, default: false },
+  itemSize: {
+    type: [Number, String] as PropType<string | number>,
+    default: 0,
+    validator: (v: number | string) => {
+      if (typeof v === 'string') { return (!isNaN(+v) || v.endsWith('px') || v.endsWith('rem')) && parseInt(v) >= 0 }
+      return v >= 0
+    },
+  },
+}
+
+export const useVirtualScrollerSizes = (
+  props: ExtractPropTypes<typeof useVirtualScrollerSizesProps>,
+  currentScroll: Ref<number>,
+) => {
+  const wrapper = shallowRef<HTMLElement>()
+  const list = shallowRef<HTMLElement>()
+
+  const itemSizeCalculated = ref(0)
+  const wrapperSize = ref(0)
+  const bodyFontSize = ref(16)
```

Comment:
ability to set sizes in refs and px should be represented in demo, also stated in docs.